src/main/java/org/cryptomator/cryptofs/health/dirid/OrphanDir.java

```diff
@@ -63,31 +63,33 @@ public void fix(Path pathToVault, VaultConfig config, Masterkey masterkey, Crypt
 		var sha1 = getSha1MessageDigest();
 		String runId = Integer.toString((short) UUID.randomUUID().getMostSignificantBits(), 32);
 		Path orphanedDir = pathToVault.resolve(Constants.DATA_DIR_NAME).resolve(this.dir);
-		String orphanHash = dir.getParent().getFileName().toString() + dir.getFileName().toString();
-
-		var stepParentDir = prepareCryptoFilesystem(pathToVault, cryptor.fileNameCryptor(), orphanHash);
-
-		try (var orphanedContentStream = Files.newDirectoryStream(orphanedDir)) {
-			AtomicInteger fileCounter = new AtomicInteger(1);
-			AtomicInteger dirCounter = new AtomicInteger(1);
-			AtomicInteger symlinkCounter = new AtomicInteger(1);
-			String longNameSuffix = createClearnameToBeShortened(config.getShorteningThreshold());
-
-			for (Path orphanedResource : orphanedContentStream) {
-				var newClearName = switch (determineCiphertextFileType(orphanedResource)) {
-					case FILE -> FILE_PREFIX + fileCounter.getAndIncrement();
-					case DIRECTORY -> DIR_PREFIX + dirCounter.getAndIncrement();
-					case SYMLINK -> SYMLINK_PREFIX + symlinkCounter.getAndIncrement();
-				} + "_" + runId;
-				adoptOrphanedResource(new Adoption(newClearName, orphanedResource), stepParentDir, cryptor.fileNameCryptor(), longNameSuffix, sha1);
+		String orphanDirIdHash = dir.getParent().getFileName().toString() + dir.getFileName().toString();
+
+		Path recoveryDir = prepareRecoveryDir(pathToVault, cryptor.fileNameCryptor());
+		if (!recoveryDir.toAbsolutePath().equals(orphanedDir.toAbsolutePath())) {
```

Comment:
I wonder if `if (equal) { return; }` might be more readable, since this is some kind of "precondition check"...